Sources/ScipioKit/Executor.swift

```diff
@@ -12,7 +12,7 @@ private actor ProcessOutputBuffer {
 @_spi(Internals)
 public protocol Executor {
     @discardableResult
-    func execute(_ arguments: [String]) async throws -> ExecutorResult
+    func execute(_ arguments: [String], environment: [String: String]?) async throws -> ExecutorResult
```

Comment:
It's not unclear from consumer-side that this argument is additional environment variables or complete replacement. The behavior should be documented.

If it's complete replacement, we should state that preserving original environment variables is caller's responsibility (by using `ProcessInfo.processInfo.environment`).